# <center>A Report on Heart Disease Dataset</center>

## 1. Data Exploration

The heart disease dataset$^{[1]}$ contains 76 attributes of which a subset of 14 attributes has been used in our dataset. The target field contains 0s and 1s with 0 representing the absence of heart disease in a patient and 1 representing the presence of it.

In [1]:
import pandas as pd

In [2]:
train_df = pd.read_csv('data/Heart_train.csv')
test_df = pd.read_csv('data/Heart_test.csv')

In [3]:
train_df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   67    1   2       152   212    0        0      150      0      0.8      1   
1   53    1   2       130   246    1        0      173      0      0.0      2   
2   61    1   3       134   234    0        1      145      0      2.6      1   
3   45    1   1       128   308    0        0      170      0      0.0      2   
4   50    1   0       144   200    0        0      126      1      0.9      1   

   ca  thal  target  
0   0     3       0  
1   3     2       1  
2   2     2       0  
3   0     2       1  
4   0     3       0

In [4]:
test_df.head()

age  sex  cp  trestbps  chol  fbs  restecg  thalach  exang  oldpeak  slope  \
0   58    1   1       120   284    0        0      160      0      1.8      1   
1   52    1   0       112   230    0        1      160      0      0.0      2   
2   42    0   2       120   209    0        1      173      0      0.0      1   
3   55    1   1       130   262    0        1      155      0      0.0      2   
4   53    0   0       130   264    0        0      143      0      0.4      1   

   ca  thal  target  
0   0     2       0  
1   1     2       0  
2   0     2       1  
3   0     2       1  
4   0     2       1

In [5]:
train_df.shape

(242, 14)

In [6]:
test_df.shape

(61, 14)

Our dataset contains a total of **242 entries in the training set** and **61 entries in the testing set**. Each dataset has **14 features**.

In [7]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 242 entries, 0 to 241
Data columns (total 14 columns):
age         242 non-null int64
sex         242 non-null int64
cp          242 non-null int64
trestbps    242 non-null int64
chol        242 non-null int64
fbs         242 non-null int64
restecg     242 non-null int64
thalach     242 non-null int64
exang       242 non-null int64
oldpeak     242 non-null float64
slope       242 non-null int64
ca          242 non-null int64
thal        242 non-null int64
target      242 non-null int64
dtypes: float64(1), int64(13)
memory usage: 26.6 KB


A brief description of the attributes is tabulated below$^{[1]}$:

S.No.|Attribute|Description
-----|:--------|:----------
1.|age|Age in years
2.|ca|Number of major vessels (0-3) colored by flourosopy
3.|chol|Serum cholestoral in mg/dl
4.|cp|Chest pain type<br><br>Value 1: typical angina<br>Value 2: atypical angina<br>Value 3: non-anginal pain<br>Value 4: asymptomatic
5.|exang|Exercise induced angina (1 = yes; 0 = no)
6.|fbs|Fasting blood sugar > 120 mg/dl (1 = true; 0 = false)
7.|oldpeak|ST depression induced by exercise relative to rest
8.|restecg|Resting electrocardiographic results<br><br>Value 0: normal<br>Value 1: having ST-T wave abnormality (T wave inversions and/or ST elevation or depression of > 0.05 mV)<br>Value 2: showing probable or definite left ventricular hypertrophy by Estes' criteria
9.|sex|1 = male; 0 = female
10.|slope|the slope of the peak exercise ST segment<br><br>Value 1: unsloping<br>Value 2: flat<br>Value 3: downsloping
11.|thal|3 = normal; 6 = fixed defect; 7 = reversable defect
12.|thalach|maximum heart rate achieved
13.|trestbps|resting blood pressure
14.|target|diagnosis of heart disease (angiographic disease status)<br><br>Value 0: < 50% diameter narrowing<br>Value 1: > 50% diameter narrowing

## References

1. https://archive.ics.uci.edu/ml/datasets/Heart+Disease